src/agent/communicator/CMakeLists.txt

```diff
@@ -10,33 +10,16 @@ set(VCPKG_MANIFEST_DIR ${CMAKE_SOURCE_DIR}/../../)
 
 project(Communicator)
 
-include_directories(include)
-
 find_package(OpenSSL REQUIRED)
-find_package(Boost REQUIRED COMPONENTS system url)
-if(Boost_FOUND)
-    include_directories(${Boost_INCLUDE_DIRS})
-    link_directories(${Boost_LIBRARY_DIRS})
-endif()
-
+find_package(Boost REQUIRED COMPONENTS asio beast)
+find_package(nlohmann_json CONFIG REQUIRED)
 find_path(JWT_CPP_INCLUDE_DIRS "jwt-cpp/base.h")
 
-add_library(communicator src/communicator.cpp)
-
-set_target_properties(communicator PROPERTIES
-LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
-
-target_include_directories(communicator PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} ${Boost_LIBRARIES} ${JWT_CPP_INCLUDE_DIRS} OpenSSL::SSL OpenSSL::Crypto)
-target_link_libraries(communicator PUBLIC ${Boost_LIBRARIES} OpenSSL::SSL OpenSSL::Crypto)
+add_library(communicator src/communicator.cpp src/http_client.cpp)
```

Comment:
Use `Communicator` name to be consisten with other libraries. Apply this change where necessary.

Reply:
done 05bf17df02fec16afd3a4954836dd16a48723f70